concent-builder/configure-jenkins.yml

```diff
@@ -0,0 +1,184 @@
+            owner: "jenkins"
+            group: "jenkins"
+          with_items:
+            - { path: "/users/admin/", name: "config.xml" }
+            - { path: "/users/admin/", name: "apiTokenStats.xml" }
+            - { path: "/", name: "credentials.xml" }
+            - { path: "/secrets/", name: "master.key" }
+            - { path: "/secrets/", name: "hudson.util.Secret" }
+
+        - name: Upload jenkins configuration file
+          # FIXME This need use as remote_src
+          copy:
+            src:   "jenkins-configuration-files/{{ item.name }}"
+            dest:  "/var/lib/jenkins{{ item.path }}{{ item.name }}"
+            owner: "jenkins"
+            group: "jenkins"
+          with_items:
+            - { path: "/jobs/Build-deploy-and-test/", name: "config.xml" }
+            - { path: "/", name: "github-plugin-configuration.xml" }
+            - { path: "/", name: "jenkins.model.JenkinsLocationConfiguration.xml" }
+
+        - name: Restart jenkins service
+          service:
+            enabled: yes
+            name:    jenkins
+            state:   restarted
+
+    - become:      yes
+      become_user: jenkins
+      block:
+        - name: Wait for jenkins after restart daemon
+          wait_for:
+            port:  8080
+            delay: 60
+
+        - name: Download jenkins-cli
+          get_url:
+            url:   "http://{{ jenkins_address }}/jnlpJars/jenkins-cli.jar"
+            dest:  "{{ jenkins_home_dir }}/jenkins-cli.jar"
+            mode:  0755
+
+        - name: Install needed plugin for jenkins
+          command: java                                               \
+              -jar jenkins-cli.jar                                    \
+              -s http://localhost:8080/                               \
+              -auth @/home/jenkins/concent-secrets/jenkins/.token     \
+              install-plugin                                          \
+              "{{ item }}"
+          args:
+            chdir: "{{ jenkins_home_dir }}/"
+          with_items:
+            - workflow-aggregator
+            - ansible
+
+        - name: Install github plugin for jenkins
+          command: java                                               \
+              -jar jenkins-cli.jar                                    \
+              -s http://localhost:8080/                               \
+              -auth @/home/jenkins/concent-secrets/jenkins/.token     \
+              install-plugin                                          \
+              github-oauth
+          args:
+            chdir: "{{ jenkins_home_dir }}/"
+
+    - become:      yes
+      become_user: root
+      block:
+        - name: Restart jenkins service
+          service:
+            enabled: yes
+            name:    jenkins
+            state:   restarted
+
+        - name: Add nginx config
+          copy:
+            src:   "jenkins-configuration-files/default"
+            dest:  "/etc/nginx/sites-enabled/default"
```

Comment:
You should put the configuration in `sites-available` and create a link to it in `sites-enabled`.

Reply:
Why do you want to complicate it in this way ?

Reply:
I think that if we want change the current solution, we could add `jenkins.conf` file to the `/etc/nginx/conf.d/` directory and delete the `/etc/nginx/sites-enabled/default` config.

Reply:
> Why do you want to complicate it in this way ?

Because that's the standard way to do it. If you use things the way they're supposed to be used, it's always easier for other people to understand your stuff. Someone familiar with how nginx works would naturally assume that `sites-enabled` contains just links and could remove the configuration file by mistake thinking it's just a link.

I think it fits more in `sites-available`/`sites-enabled`. That's where you're supposed to put your `server` blocks. `conf.d` sounds like something where you would put bits of global configuration that pertain to all the sites

Reply:
It's still not fixed. The file is still in `sites-enabled`.